workflow/executor/agent.go

```diff
@@ -265,14 +265,25 @@ func (ae *AgentExecutor) executeHTTPTemplate(ctx context.Context, tmpl wfv1.Temp
 			message = fmt.Sprintf("received non-2xx response code: %d", response.StatusCode)
 		}
 	} else {
-		evalScope := map[string]interface{}{
-			"request": map[string]interface{}{
+		var request map[string]interface{}
+		if tmpl.HTTP.BodyFrom != nil && tmpl.HTTP.BodyFrom.Bytes != nil {
+			request = map[string]interface{}{
 				"method":    tmpl.HTTP.Method,
 				"url":       tmpl.HTTP.URL,
 				"body":      tmpl.HTTP.Body,
 				"bodyBytes": tmpl.HTTP.BodyFrom.Bytes,
 				"headers":   tmpl.HTTP.Headers.ToHeader(),
-			},
+			}
+		} else {
+			request = map[string]interface{}{
```

Comment:
This is code duplication. I think you add `tmpl.HTTP.GetBodyBytes()` which can return `nil` when `tmpl.BodyFrom` is nil